oracle/pkg/apiserver/apiserver.go

```diff
@@ -71,10 +86,81 @@ func New(
 		logger.Error("failed to configure dashboard", "error", err)
 	}
 
+	srv.router.Handle("/register_provider", srv.registerProvider(password))
+
 	srv.registerDebugEndpoints()
 	return srv
 }
 
+func (s *Service) registerProvider(password string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		if r.Header.Get("X-Registration-Password") != password {
```

Comment:
How do we plan to use this password? I don’t see that it is saved or used later in the code.